lib/components/base-components/NormalComponent.ts

```diff
@@ -115,6 +122,26 @@ export class NormalComponent<
           }
         }
       }
+      // pin label are string
+      if (normalizationNeeded) {
+        const normalizedPinLabels =
+          this._getNormalizedPinLabelsFromSchPortArrangement()
+        for (const [pinLabel, pinName] of Object.entries(normalizedPinLabels)) {
+          const pinNumber = parseInt(pinLabel.replace("pin", ""))
+          portsToCreate.push(
+            new Port(
+              {
+                pinNumber,
+                name: pinName.toString(),
+                aliases: [pinLabel, pinName],
+              },
+              {
+                originDescription: `schPortArrangement:pin${pinNumber}`,
+              },
+            ),
+          )
+        }
+      }
```

Comment:
this code seems to create duplicate ports. You're them before you know if normalization is needed